src/main/java/org/example/homework03/repository/UserRepositoryImpl.java

```diff
@@ -0,0 +1,103 @@
+    private static final String PASSWORD = "password";
+
+    public User persistUser(User user) {
+        String sql = "INSERT INTO users (email, phoneNumber, password) VALUES (?, ?, ?);";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+            PreparedStatement preparedStatement = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, user.getEmail());
+            preparedStatement.setString(2, user.getPhoneNumber());
+            preparedStatement.setString(3, user.getPassword());
+
+            int affectedRows = preparedStatement.executeUpdate();
+            checkAffectedRowsNumber(affectedRows);
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                user.setId(id);
+            }
+
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+
+        return user;
+    }
+
+    public Optional<User> getUserById(Long id) {
+        String sql = "SELECT * FROM users WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(parseUser(resultSet));
+            }
+
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+
+        return Optional.empty();
+    }
+
+    public Optional<User> getUserByEmail(String email) {
+        String sql = "SELECT * FROM users WHERE email = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            preparedStatement.setString(1, email);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(parseUser(resultSet));
+            }
+
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+
+        return Optional.empty();
+    }
+
+    private User parseUser(ResultSet resultSet) {
+        try {
+            Long id = resultSet.getObject(ID, Long.class);
+            String email = resultSet.getString(EMAIL);
+            String phoneNumber = resultSet.getString(PHONE_NUMBER);
+            String password = resultSet.getString(PASSWORD);
+
+            return new User(
+                    id,
+                    email,
+                    phoneNumber,
+                    password
+            );
+        } catch (SQLException e) {
+            throw new RuntimeException("Can not parse 'user' from resultSet.");
```

Comment:
re-throw the original error to have original stack trace for better debugging
```suggestion
            throw new RuntimeException("Can not parse 'user' from resultSet.", e);
```